src/flow.jl

```diff
@@ -38,29 +37,44 @@ abstract type SurfaceFlow end
-    h::Vector{T} | "m"                              # Water level [m]
-    h_av::Vector{T} | "m"                           # Average water level [m]
-    dt::T | "s" | 0 | "none" | "none"               # Model time step [s]
-    its::Int | "-" | 0 | "none" | "none"            # Number of fixed iterations
-    width::Vector{T} | "m"                          # Flow width [m]
-    alpha_pow::T | "-" | 0 | "scalar"               # Used in the power part of alpha
-    alpha_term::Vector{T} | "-"                     # Term used in computation of alpha
-    alpha::Vector{T} | "s3/5 m1/5"                  # Constant in momentum equation A = alpha * Q^beta, based on Manning's equation
-    cel::Vector{T} | "m s-1"                        # Celerity of the kinematic wave
-    to_river::Vector{T} | "m3 s-1"                  # Part of overland flow [m³ s⁻¹] that flows to the river
-    kinwave_it::Bool | "-" | 0 | "none" | "none"    # Boolean for iterations kinematic wave
+@get_units @with_kw struct SurfaceFlowLand{T} <: SurfaceFlow
+    beta::T | "-"                       # constant in Manning's equation
+    sl::Vector{T} | "m m-1"             # Slope [m m⁻¹]
+    n::Vector{T} | "s m-1/3"            # Manning's roughness [s m⁻⅓]
+    dl::Vector{T} | "m"                 # Drain length [m]
+    q::Vector{T} | "m3 s-1"             # Discharge [m³ s⁻¹]
+    qin::Vector{T} | "m3 s-1"           # Inflow from upstream cells [m³ s⁻¹]
+    q_av::Vector{T} | "m3 s-1"          # Average discharge [m³ s⁻¹]
+    qlat::Vector{T} | "m2 s-1"          # Lateral inflow per unit length [m² s⁻¹]
+    inwater::Vector{T} | "m3 s-1"       # Lateral inflow [m³ s⁻¹]
+    volume::Vector{T} | "m3"            # Kinematic wave volume [m³] (based on water level h)
+    h::Vector{T} | "m"                  # Water level [m]
+    h_av::Vector{T} | "m"               # Average water level [m]
+    dt::T | "s"                         # Model time step [s]
+    its::Int | "-"                      # Number of fixed iterations
+    width::Vector{T} | "m"              # Flow width [m]
+    alpha_pow::T | "-"                  # Used in the power part of alpha
+    alpha_term::Vector{T} | "-"         # Term used in computation of alpha
+    alpha::Vector{T} | "s3/5 m1/5"      # Constant in momentum equation A = alpha * Q^beta, based on Manning's equation
+    cel::Vector{T} | "m s-1"            # Celerity of the kinematic wave
+    to_river::Vector{T} | "m3 s-1"      # Part of overland flow [m³ s⁻¹] that flows to the river
+    kinwave_it::Bool | "-"              # Boolean for iterations kinematic wave
+end
+
+function exchange(::SurfaceFlow, var)
+    if var in (:dt, :beta, :its, :alpha_pow, :reservoir, :lake, :kinwave_it)
+        0
+    else
+        1
+    end
+end
+
+function grid_location(::SurfaceFlow, var)
+    if var in (:dt, :its, :kinwave_it)
+        "none"
+    else
+        "node"
+    end
```

Comment:
I'm not too familiar with these functions, but it feels like they are quite specific to BMI: is that right? Is it maybe better to keep track of these in the src/bmi.jl file, or do you prefer to keep it close to their source? If it is quite specific BMI information/functionality, maybe it is better to keep those functions in the scr/bmi.jl file (a bit similar to the src/states.jl functionality). If it is not specific for BMI, keeping them here is fine with me!

Reply:
Yes, these functions are quite specific to BMI. I agree and will move these functions to src/bmi.jl.